README.md

```diff
@@ -31,6 +31,130 @@ Application-level tracing for Rust.
+        return Err(io::Error::new(io::ErrorKind::Other, "shaving yak failed!").into());
+    } else {
+        debug!("yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut yaks_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        debug!(yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(yak, error = error.as_ref(), "failed to shave yak!");
+        } else {
+            yaks_shaved += 1;
+        }
+        debug!(yaks_shaved);
+    }
+
+    yaks_shaved
+}
+```
+
+### In applications
+
+In order to produce output, applications need a concrete subscriber implementation. 
+[`tracing_subscriber`](https://docs.rs/tracing-subscriber/) is a reasonable default.
+By default, `tracing-subscriber` is able to consume messages emitted by
+`log`-instrumented libraries and modules.
+
+```toml
+[dependencies]
+tracing = "0.1"
+tracing-subscriber = "0.2.0-alpha.2"
+```
+
+```rust
+use tracing::{info, Level};
+use tracing_subscriber;
+mod yak_shave;
+
+fn main() {
+    tracing_subscriber::fmt::Subscriber::builder()
+        // all spans/events with a level higher than `DEBUG` (e.g, info, warn, etc.)
+        // will be written to stdout.
+        .with_max_level(Level::DEBUG)
+        // sets this to be the default, global subscriber for this application.
+        .init();
+
+    let number_of_yaks = 3;
+    info!(number_of_yaks, "preparing to shave yaks");
+
+    let number_shaved = yak_shave::shave_all(number_of_yaks);
+    info!(
+        all_yaks_shaved = number_shaved == number_of_yaks,
+        "yak shaving completed."
+    );
+}
+```
+
+### Additional Notes
+
+#### Usage in Asynchronous Code
+
+If you are instrumenting libraries or applications that make use of [`std::future::Future`](https://doc.rust-lang.org/stable/std/future/trait.Future.html) or async/await, be sure to use the [`tracing-futures`](https://docs.rs/tracing-futures) crate. There are two ways to instrument asynchronous code. The first is through the [`#[instrument]`](https://docs.rs/tracing/0.1.11/tracing/attr.instrument.html) attribute:
+
+```rust
+use tracing::{info, instrument};
+use tokio::{io::AsyncWriteExt, net::TcpStream};
+use std::io;
+
+#[instrument]
+async fn write(stream: &mut TcpStream) -> io::Result<usize> {
+    let result = stream.write(b"hello world\n").await;
+    info!("wrote to stream; success={:?}", result.is_ok());
+    result
+}
+```
+
+The second is through the [`.instrument`](https://docs.rs/tracing-futures/0.2.0/tracing_futures/trait.Instrument.html#method.instrument) combinator:
+
+```rust
+use tracing_futures::Instrument;
+
+let my_future = async {
+    // ...
+};
+
+my_future
+    .instrument(tracing::info_span!("my_future"))
+    .await
+```
```

Comment:
Can we have a very brief explanation of what these examples do and why you shouldn't just write
```rust
async {
    let _s = span.enter();
    // ...
}
```

Since this is a common source of confusion block for new users, I'd really like to make sure folks understand this.